### PR TITLE
Adding revision to Environment resource

### PR DESCRIPTION
**Pull request by IaroslavTitov**
### Summary
- Adding a revision property to Environment resource
- Moving to a newer esc client added more dummy methods into the test file

### Testing
- Tested in devstack using Dotnet API
- revision number changes as expected

### CODE REVIEW

File: provider/pkg/provider/environment_test.go [hunk 1]
```diff
@@ -14,14 +14,26 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-type getEnvironmentFunc func(ctx context.Context, orgName string, envName string, decrypt bool) (yaml []byte, etag string, err error)
+type getEnvironmentFunc func(ctx context.Context, orgName string, envName string, version string, decrypt bool) (yaml []byte, etag string, revision int, err error)
```
Comment (komalali) [inline provider/pkg/provider/environment_test.go:17]: What is the `version` here?

File: sdk/nodejs/environment.ts [hunk 2]
```diff
@@ -70,9 +74,11 @@ export class Environment extends pulumi.CustomResource {
             resourceInputs["name"] = args ? args.name : undefined;
             resourceInputs["organization"] = args ? args.organization : undefined;
             resourceInputs["yaml"] = args ? args.yaml : undefined;
+            resourceInputs["revision"] = undefined /*out*/;
```
Comment (komalali) [inline sdk/nodejs/environment.ts:77]: This is odd... why is it in the inputs section?

File: provider/pkg/provider/environment_test.go (hunk 1 above)
Comment (IaroslavTitov) [reply provider/pkg/provider/environment_test.go:17]: This function has to match the signature of GetEnvironment method inside ESC client. I bumped the version and signature of the method changed, so had to change to match as well.

File: sdk/nodejs/environment.ts (hunk 2 above)
Comment (komalali) [reply sdk/nodejs/environment.ts:77]: I think this is fine, I'm just not used to seeing outputs only.

**Review by komalali — APPROVED** [review]

File: sdk/nodejs/environment.ts (hunk 2 above)
Comment (IaroslavTitov) [reply sdk/nodejs/environment.ts:77]: I'm guessing because it's an output only property?

**Pull request merged**